packit/cli/builds/obs_build.py

```diff
@@ -0,0 +1,273 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os.path
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Optional
+
+import click
+from osc import conf, core
+
+from packit.cli.types import LocalProjectParameter
+from packit.cli.utils import cover_packit_exception, get_packit_api, iterate_packages
+from packit.config import get_context_settings, pass_config
+from packit.config.aliases import DEPRECATED_TARGET_MAP
+from packit.config.config import Config
+from packit.config.package_config import PackageConfig
+from packit.constants import (
+    PACKAGE_LONG_OPTION,
+    PACKAGE_OPTION_HELP,
+    PACKAGE_SHORT_OPTION,
+)
+
+logger = logging.getLogger(__name__)
+
+_API_URL = "https://api.opensuse.org"
```

Comment:
Are there more instances? Is there a chance that something needs to connect an alternative one?

Just asking if we should have it in `constants.py`  or in the user/service config.

Note that we try to use prod instances of external services for all packit instances (i.e. stage) so we rather verify Packit and not other services.

Reply:
The other one I know is OpenHPC: https://obs.openhpc.community/

This project + testing farm would actually be a good idea to introduce to them

Reply:
It is not that easy. The biggest strength of OBS is also its biggest weakness: it is customizable beyond any other build system. Currently the code relies on the configuration of build.opensuse.org, e.g. that the project Fedora:Rawhide contains the repositories that packit expects to get for the target `fedora-rawhide`. Another OBS instance might not have this project at all, or it can be a totally broken playground, or something entirely different.

As an ordinary user, you cannot even configure the projects required for packit support. You would have to create so called download-on-demand repositories, which afaik require admin permissions.

Hence, I would for now strongly suggest to **not** make the OBS instance configurable at the moment. For instance the proposed openhpc instance lacks all openSUSE, SLE and Fedora projects and the CentOS setup is also different from the one on OBS.

Reply:
Couldn't it be introspected via the rest api? Otherwise it could allow to `try ... catch` the commands with the pure string in the targets, without supporting aliases or anything specific. Forthe known configuration like openSUSE's OBS it should do the checks and validations early on.

Reply:
It is really **not** that simple. You can check the direct repository configuration, but you'd actually need to check all recursively included projects **and** their `prjconf` (this is a side-macro configuration) just to be sure. And things could then still probably break. Honestly, this is a nightmare that I would prefer not to go down, at least not yet.

Reply:
@dcermak Thanks for the clarification. Looks like we can really start with a single instance and add others if really needed. (Just use constants where meaningful.)